commit dd28a4b0a02e93488d8849f1579379ce26ee9afe (refs/changes/40/2840/1)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-10-23T09:54:56+02:00 (2 years ago)
    
    TOOLS-1902 'sdcadm experimental up -s UUID' should validate provided UUIDs better

diff --git a/lib/procedures/index.js b/lib/procedures/index.js
index 94780c0..b413770 100644
--- a/lib/procedures/index.js
+++ b/lib/procedures/index.js
@@ -692,6 +692,14 @@ function coordinatePlan(opts, cb) {
             // Make sure we get a list of uuids even if hostnames may be
             // initially present as servers option:
             if (opts.servers && opts.servers.length) {
+                var notFound = opts.servers.filter(function (s) {
+                    return (!opts.serverFromUuidOrHostname[s]);
+                });
+                if (notFound.length) {
+                    next(new errors.UpdateError(format(
+                            'unknown SDC servers "%s"', notFound.join(', '))));
+                    return;
+                }
                 opts.servers = opts.servers.map(function (s) {
                     return opts.serverFromUuidOrHostname[s].uuid;
                 });
